### PR TITLE
Style: ssl.h

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
fix some indents, and restrict to 80 cols some lines.

split from #4457 .

##### Checklist
- [X] documentation is added or updated
- [N/A] tests are added or updated

### CODE REVIEW

File: include/openssl/ssl.h [hunk 1]
```diff
@@ -217,11 +217,9 @@ typedef struct srtp_protection_profile_st {
 
 DEFINE_STACK_OF(SRTP_PROTECTION_PROFILE)
 
-typedef int (*tls_session_ticket_ext_cb_fn) (SSL *s,
-                                             const unsigned char *data,
+typedef int (*tls_session_ticket_ext_cb_fn) (SSL *s, const unsigned char *data,
```
Comment (kaduk) [inline include/openssl/ssl.h]: A style pass over ssl.h might reasonably be expected to normalize on one of ') (' or ')(' in function typedefs.
https://github.com/openssl/openssl/pull/2279#discussion_r102722095 is perhaps in support of the latter (no space), but we've really been all over the place.

File: include/openssl/ssl.h [hunk 2]
```diff
@@ -688,8 +685,7 @@ SSL_SESSION *(*SSL_CTX_sess_get_get_cb(SSL_CTX *ctx)) (struct ssl_st *ssl,
                                                        const unsigned char *data,
```
Comment (kaduk) [inline include/openssl/ssl.h]: ',' is in column 81

File: include/openssl/ssl.h [hunk 3]
```diff
@@ -1817,7 +1821,7 @@ DEPRECATEDIN_1_1_0(__owur const SSL_METHOD *DTLSv1_client_method(void)) /* DTLSv
 DEPRECATEDIN_1_1_0(__owur const SSL_METHOD *DTLSv1_2_method(void)) /* DTLSv1.2 */
 DEPRECATEDIN_1_1_0(__owur const SSL_METHOD *DTLSv1_2_server_method(void)) /* DTLSv1.2 */
 DEPRECATEDIN_1_1_0(__owur const SSL_METHOD *DTLSv1_2_client_method(void)) /* DTLSv1.2 */
```
Comment (kaduk) [inline include/openssl/ssl.h]: Most of these deprecated version-specific methods have their comment extend past the 80th column

File: include/openssl/ssl.h [hunk 4]
```diff
@@ -2009,12 +2014,11 @@ int SSL_bytes_to_cipher_list(SSL *s, const unsigned char *bytes, size_t len,
 __owur int SSL_set_session_ticket_ext(SSL *s, void *ext_data, int ext_len);
 
 __owur int SSL_set_session_ticket_ext_cb(SSL *s, tls_session_ticket_ext_cb_fn cb,
```
Comment (kaduk) [inline include/openssl/ssl.h]: column 81

File: include/openssl/ssl.h [hunk 5]
```diff
@@ -2233,21 +2237,21 @@ const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx);
...
 void SSL_set_security_callback(SSL *s,
-                               int (*cb) (const SSL *s, const SSL_CTX *ctx, int op,
-                                          int bits, int nid, void *other,
-                                          void *ex));
-int (*SSL_get_security_callback(const SSL *s)) (const SSL *s, const SSL_CTX *ctx, int op,
-                                                int bits, int nid,
+                               int (*cb) (const SSL *s, const SSL_CTX *ctx,
+                                          int op, int bits, int nid,
+                                          void *other, void *ex));
+int (*SSL_get_security_callback(const SSL *s)) (const SSL *s, const SSL_CTX *ctx,
```
Comment (kaduk) [inline include/openssl/ssl.h]: column 81

File: include/openssl/ssl.h [hunk 6]
```diff
@@ -2009,12 +2014,11 @@ int SSL_bytes_to_cipher_list(SSL *s, const unsigned char *bytes, size_t len,
...
 
 __owur int SSL_set_session_ticket_ext_cb(SSL *s, tls_session_ticket_ext_cb_fn cb,
-                                  void *arg);
+                                         void *arg);
 
 /* Pre-shared secret session resumption functions */
-__owur int SSL_set_session_secret_cb(SSL *s,
-                              tls_session_secret_cb_fn tls_session_secret_cb,
-                              void *arg);
+__owur int SSL_set_session_secret_cb(SSL *s, tls_session_secret_cb_fn tls_session_secret_cb,
```
Comment (kaduk) [inline include/openssl/ssl.h]: Column 92(!)

**Comment by FdaSilvaYY** [comment]
I was just trying to push attention on the various nits in this file.
Can you push these PR as it is ?
Else I will fix (or continue it) in a few weeks, as I'm currently far far away ...  ;)

**Comment by kaduk** [comment]
I couldn't push anything without an OMC code review.  And this seems sufficiently non-urgent that continuing when you get back should be okay.

File: include/openssl/ssl.h (hunk 1 above)
Comment (richsalz) [reply include/openssl/ssl.h]: I would remove the space.

File: include/openssl/ssl.h [hunk 7]
```diff
@@ -253,15 +251,15 @@ typedef int (*tls_session_secret_cb_fn) (SSL *s, void *secret,
 /* Typedefs for handling custom extensions */
 
 typedef int (*custom_ext_add_cb) (SSL *s, unsigned int ext_type,
```
Comment (richsalz) [inline include/openssl/ssl.h]: and remove the spaces between the function name and the param lists here, and below, as well.

File: include/openssl/ssl.h [hunk 8]
```diff
@@ -674,8 +672,7 @@ int (*SSL_CTX_sess_get_new_cb(SSL_CTX *ctx)) (struct ssl_st *ssl,
 void SSL_CTX_sess_set_remove_cb(SSL_CTX *ctx,
                                 void (*remove_session_cb) (struct ssl_ctx_st
```
Comment (richsalz) [inline include/openssl/ssl.h]: all those function typedefs, and there isn't one here? :)

Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: I get ithese kind of changes on my git repo, as draft for a next PR.

File: include/openssl/ssl.h (hunk 2 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: better fixed with a typedef on call_proc type.

File: include/openssl/ssl.h (hunk 1 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: fixed

File: include/openssl/ssl.h (hunk 3 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: Not the most prominent style issue.

File: include/openssl/ssl.h (hunk 5 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: Need an `typedef `on callback type to fully fix it ; in a next PR...

File: include/openssl/ssl.h (hunk 7 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: Too much too fix ;)

**Comment by FdaSilvaYY** [comment]
Fixed and rebased to current master.
Ping @openssl for review ; it is only style nits, no code change.

**Comment by kaduk** [comment]
I'm not sure how much sense it makes to take a style change that is making changes to idioms not yet codified in the style guide (e.g., no space between return-type parens and function-arg parens for function typedefs).  So maybe this should move to post-1.1.1?

**Comment by kaduk** [comment]
> I'm not sure how much sense it makes to take a style change that is making changes to idioms not yet codified in the style guide (e.g., no space between return-type parens and function-arg parens for function typedefs). So maybe this should move to post-1.1.1?

Okay, that is extremely funny given that I am the one who asked for the change in the first place.  I guess I should look at the rest of the changes now

File: include/openssl/ssl.h [hunk 9]
```diff
@@ -2184,7 +2191,7 @@ void SSL_CTX_set0_ctlog_store(SSL_CTX *ctx, CTLOG_STORE *logs);
  */
 const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx);
 
-# endif /* OPENSSL_NO_CT */
+# endif                         /* OPENSSL_NO_CT */
```
Comment (kaduk) [inline include/openssl/ssl.h]: This seems to be the only comment on ```#endif``` in the file; I almost wonder if it is "more consistent" with the current style to remove it than to reindent it.

Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: This change comes from the `openssl-format-source` tool. I don't know the reason/rule behind this change.

Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: @kaduk: I will be glad to remove it, if you say so.
On the other side, I'm for having this kind of comment when the length of this block exceeds the height of the screen ;)
I rebase these changes on a fresh master .

Comment (mattcaswell) [reply include/openssl/ssl.h]: I think the comment should remain. I don't know why the tool reindented it. I think it looks better without the reindentation.

Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: Fixed and rebased to last master.

**Comment by mattcaswell** [comment]
@kaduk  any more comments?

**Comment by kaduk** [comment]
I probably won't get a chance to look again this week; let me dismiss my review at least for now.

**Comment by FdaSilvaYY** [comment]
@kaduk any more comments, or this PR is good to go with its two approvals ?

File: include/openssl/ssl.h [hunk 10]
```diff
@@ -2277,21 +2284,22 @@ const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx);
...
-                                          int bits, int nid, void *other,
-                                          void *ex));
-int (*SSL_get_security_callback(const SSL *s)) (const SSL *s, const SSL_CTX *ctx, int op,
-                                                int bits, int nid,
-                                                void *other, void *ex);
+                               int (*cb) (const SSL *s, const SSL_CTX *ctx,
+                                          int op, int bits, int nid,
+                                          void *other, void *ex));
+int (*SSL_get_security_callback(const SSL *s)) (const SSL *s,
+                                                const SSL_CTX * ctx, int op,
```
Comment (kaduk) [inline include/openssl/ssl.h]: We can nestle the ```*ctx``` with no space between

File: include/openssl/ssl.h [hunk 11]
```diff
@@ -2277,21 +2284,22 @@ const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx);
...
 void SSL_set0_security_ex_data(SSL *s, void *ex);
 __owur void *SSL_get0_security_ex_data(const SSL *s);
 
 void SSL_CTX_set_security_level(SSL_CTX *ctx, int level);
 __owur int SSL_CTX_get_security_level(const SSL_CTX *ctx);
 void SSL_CTX_set_security_callback(SSL_CTX *ctx,
-                                   int (*cb) (const SSL *s, const SSL_CTX *ctx, int op,
-                                              int bits, int nid, void *other,
-                                              void *ex));
+                                   int (*cb) (const SSL *s, const SSL_CTX *ctx,
```
Comment (kaduk) [inline include/openssl/ssl.h]: I forget if the tooling complains if we nestle these together as ```int (*cb)(const```

File: include/openssl/ssl.h (hunk 10 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: Fixed !

File: include/openssl/ssl.h (hunk 11 above)
Comment (FdaSilvaYY) [reply include/openssl/ssl.h]: I don't remember, I keep the original style; will be fixed later with a few typedef's function-pointer .

**Comment by FdaSilvaYY** [comment]
Ping @richsalz , @mattcaswell : little nudge for reconfirming and merge ;)

**Comment by richsalz** [comment]
@mattcaswell you want to reconfirm, and possibly merge?

**Comment by FdaSilvaYY** [comment]
Ping @richsalz , @mattcaswell this looks like it is lost in the limbo ? ;)

**Comment by mattcaswell** [comment]
Sorry for the delay. Pushed now. Thanks.

**Pull request closed without merging**